swap_meet/vendor.py

```diff
@@ -1,2 +1,91 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        first_item = self.inventory[0]
+        first_item_other = other_vendor.inventory[0]
+        if  first_item  and first_item_other:
+            self.inventory.remove(first_item)
+            other_vendor.inventory.append(first_item)
+            other_vendor.inventory.remove(first_item_other)
+            self.inventory.append(first_item_other)
+            return True
+    
+    def get_by_category(self, category="Unknown"):
+        return[item for item in self.inventory if item.get_category() == category]
+    
+    def get_highest_item(self, list_items, key=lambda x: x):
```

Comment:
You don't need `x` or `lambda` in the function definition here. You do need the lambda key word and you do need to declare `item` as an argument for the lambda expression on line 62, but for this function we just need `key` in the params list to use it.

You iterate over `item` in `list_items`.

```suggestion
    def get_highest_item(self, list_items, key):
```

Reply:
✅ Fixed (will push later!)